cmd/zoekt-sourcegraph-indexserver/main.go

```diff
@@ -1135,14 +1142,42 @@ func newServer(conf rootConfig) (*Server, error) {
 }
 
 func main() {
+	cmd := rootCmd()
+	if err := cmd.Parse(os.Args[1:]); err != nil {
+		log.Fatal(err)
+	}
+
+	// Visit the flags to determine if debug flag is included and enabled
+	debugEnabled := false
+	cmd.FlagSet.Visit(func(flag *flag.Flag) {
+		if flag.Name == "debug" {
+			if toBool, err := strconv.ParseBool(flag.Value.String()); err == nil {
+				debugEnabled = toBool
+			}
+		}
+	})
+
 	name, _ := findName()
+	originalLogLevel := os.Getenv(srcLogLevel)
+
+	// if debug flag is included and enabled then modify srcLogLevel
+	// for logger's Init(). Otherwise rely on srcLogLevel original value.
+	if debugEnabled {
+		os.Setenv(srcLogLevel, "debug")
+	}
+
 	syncLogs := sglog.Init(sglog.Resource{
 		Name:       name,
 		Version:    zoekt.Version,
 		InstanceID: hostnameBestEffort(),
 	})
+
+	os.Setenv(srcLogLevel, originalLogLevel)
```

Comment:
Also, this must be before `sglog.Init`

Reply:
This is the reverting of the environment variable to the original value so that later calls to `srcLogLevelIsDebug` use the original value

Reply:
We only `Setenv` to debug if the flag is passed in